lib/stimulus_reflex/reflex.rb

```diff
@@ -26,32 +12,26 @@ class StimulusReflex::Reflex
   include CableReady::Identifiable
 
   attr_accessor :payload, :headers
-  attr_reader :channel, :url, :element, :selectors, :method_name, :broadcaster, :client_attributes, :logger
-
-  alias_method :action_name, :method_name # for compatibility with controller libraries like Pundit that expect an action name
+  attr_reader :channel, :reflex_data, :broadcaster
 
   delegate :connection, :stream_name, to: :channel
   delegate :controller_class, :flash, :session, to: :request
   delegate :broadcast, :broadcast_halt, :broadcast_forbid, :broadcast_error, to: :broadcaster
+
   # TODO remove xpath_controller and xpath_element for v4
-  delegate :id, :tab_id, :reflex_controller, :xpath_controller, :xpath_element, :permanent_attribute_name, :version, :npm_version, :suppress_logging, to: :client_attributes
+  delegate :url, :element, :selectors, :method_name, :id, :tab_id, :reflex_controller, :xpath_controller, :xpath_element, :permanent_attribute_name, :version, :npm_version, :suppress_logging, to: :reflex_data
+  # END TODO: remove
 
-  def initialize(channel, url: nil, element: nil, selectors: [], method_name: nil, params: {}, client_attributes: {})
+  alias_method :action_name, :method_name # for compatibility with controller libraries like Pundit that expect an action name
+
+  def initialize(channel, reflex_data:)
     @channel = channel
-    @url = url
-    @element = element
-    @selectors = selectors
-    @method_name = method_name
-    @params = params
-    @client_attributes = ClientAttributes.new(client_attributes)
+    @reflex_data = reflex_data
     @broadcaster = StimulusReflex::PageBroadcaster.new(self)
-    @logger = suppress_logging ? nil : StimulusReflex::Logger.new(self)
     @payload = {}
     @headers = {}
 
     check_version!
-
-    self.params
```

Comment:
I don't have time to make a reproduction, but I believe that removing the `self.params` call will bring back a bug, where the params, may not be set once the `request` reaches the action controller. 

Calling the self.params ensures that the params are properly set in the request object / headers when passing the request to the ActionController to rerender the current action. see https://github.com/stimulusreflex/stimulus_reflex/pull/249

Reply:
I appreciate your feedback, but I could not find any confirmation of your theory in the code.

I have [updated a test](https://github.com/stimulusreflex/stimulus_reflex/pull/663/files#diff-7e418500360dd99316b785f57d0ac7ef7d79c17d855fcefe2da575cd0d8d2492R36-R44) to remove unnecessary stubbing and better exercise the code around request parameters. If I were to print out `reflex.controller.request.parameters` there right before the first assertion, it would output `{"controller"=>"test", "action"=>"index", "a"=>"1", "b"=>"2", "c"=>"3"}`, which confirms that all parameters do get passed along to the controller instance without the `self.params` in the constructor.

Reply:
@alexander-makarenko ah you are right, I made a quick test app, and it indeed worked. 

Request params are a bit weird, if you set the at the wrong moment they won't persist as the request memoizes them. But apprently this new implementation works around that.

So all is good! Thank you!